Add placeholder image for product preview

coderkevin opened the pull request:
This adds the designated placeholder image url from settings into the
wcSettings global for the blocks to use. Then uses the placeholder url
when a product doesn't have any images.

Fixes #349

#### Accessibility

<!-- If you've changed or added any interactions, check off the appropriate items below. You can delete any that don't apply. Use this space to elaborate on anything if needed. -->

- [x] I've tested using only a keyboard (no mouse)
- [x] I've tested using a screen reader
- [x] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots

<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->

![edit_post_ _two_wordpress_test_ _wordpress](https://user-images.githubusercontent.com/945228/52022974-e6710c80-24c0-11e9-9941-5d190f1daa52.png)

### How to test the changes in this Pull Request:

1. Ensure you have a placeholder set in WooCommerce -> Settings -> Products
2. Create a new block that has a product in it that you know doesn't have a product picture.
3. You should see the designated placeholder image.

<!-- If you can, add the appropriate labels -->

coderkevin committed:
Add placeholder image for product preview

This adds the designated placeholder image url from settings into the
wcSettings global for the blocks to use. Then uses the placeholder url
when a product doesn't have any images.

ryelle commented on woocommerce-gutenberg-products-block.php:
```diff
@@ -375,19 +375,20 @@ function wgpb_print_script_settings() {
-		'l10n'          => array(
+		'stockStatuses'     => wc_get_product_stock_status_options(),
+		'siteTitle'         => get_bloginfo( 'name' ),
+		'placeholderImgSrc' => wc_placeholder_img_src(),
```
This should be added to the `$block_settings` array just below this, which is used for the `wc_product_block_data` global. We don't use `wcSettings` directly in this project, it's just here to not cause errors with using `@woocommerce/components`

ryelle commented on assets/js/components/product-preview/index.js:
```diff
@@ -13,9 +13,16 @@ import './style.scss';
+	} else {
+		image = <img
+			src={ placeholderImgSrc }
+			alt={ __( 'Placeholder Image', 'woo-gutenberg-products-block' ) }
```
This alt text should be an empty string– a placeholder image is just visual decoration, so we don't need to add information for screen readers.

ryelle reviewed (changes requested):
Looks good in the block preview, but needs a few code changes. The ProductPreview test is also failing now, can you update the snapshot there? You can run the test locally with `npm test`, and recreate the snapshot with `npm run test:update`

coderkevin committed:
Updated based on review notes

Using `$block_settings` instead of `$settings`
Using '' alt tag
Updated snapshot

coderkevin commented:
Thanks for the review, @ryelle. I've updated the PR and it's ready for review again.

ryelle reviewed (dismissed):
One last change request for the test, then it should be good to merge 🙂

ryelle committed:
Add the placeholder global to test that it passes through to the preview
